modules/bmc/bmc_api.rb

```diff
@@ -1,30 +1,48 @@
+require 'bmc/ipmi'
```

Comment:
You are adding global require here, this will cause issues on discovery-image. Please load dependencies only when ipmi is enabled by configuration later on, not here.